src/App.js

```diff
@@ -1,58 +1,45 @@
-import React from 'react'
+import React, { useState } from 'react'
+
 import FakeSearchAPI from './fake-api/search'
 import { PAutoComplete } from './components/widgets'
+import { debounce } from './utils'
 import './App.css'
 
-class App extends React.Component {
-  constructor(props) {
-    super(props)
-    this.state = {
-      keyword: '',
-      results: [],
-    }
-  }
-  updateKeyword(keyword) {
-    return new Promise((resolve) => {
-      this.setState(
-        {
-          keyword,
-        },
-        () => resolve(keyword)
-      )
-    })
-  }
-  updateResult(results) {
-    return new Promise((resolve) => {
-      this.setState(
-        {
-          results,
-        },
-        () => resolve()
-      )
-    })
-  }
-  async search(keyword) {
+const App = () => {
+  const [loading, setLoading] = useState(false)
+  const [suggestions, setSuggestions] = useState([])
+
+  const search = debounce(async (keyword) => {
```

Comment:
Using `debounce` utility don't really fix the issue `KI-01`. It can only avoid most of the case (like 70%).